extensions/vertx-web/runtime/src/main/java/io/quarkus/vertx/web/runtime/cors/CORSFilter.java

```diff
@@ -20,34 +25,43 @@ public CORSFilter(CORSConfig corsConfig) {
         this.corsConfig = corsConfig;
     }
 
-    private void processRequestedHeaders(HttpServerResponse response, String requestedHeaders) {
+    private void processRequestedHeaders(HttpServerResponse response, String allowHeadersValue) {
         if (corsConfig.headers.isEmpty()) {
-            response.headers().set(HttpHeaders.ACCESS_CONTROL_ALLOW_HEADERS, requestedHeaders);
+            response.headers().set(HttpHeaders.ACCESS_CONTROL_ALLOW_HEADERS, allowHeadersValue);
         } else {
-            final String validRequestHeaders = corsConfig.headers
-                    .stream()
-                    .filter(requestedHeaders::contains)
-                    .collect(Collectors.joining(","));
+            List<String> requestedHeaders = new ArrayList<>();
+            for (String requestedHeader : allowHeadersValue.split(COMMA_SEPARATED_SPLIT_REGEX)) {
+                requestedHeaders.add(requestedHeader.toLowerCase());
+            }
 
-            if (!validRequestHeaders.isEmpty()) {
-                response.headers().set(HttpHeaders.ACCESS_CONTROL_ALLOW_HEADERS, validRequestHeaders);
+            List<String> validRequestedHeaders = new ArrayList<>();
+            for (String configHeader : corsConfig.headers) {
+                if (requestedHeaders.contains(configHeader.toLowerCase())) {
+                    validRequestedHeaders.add(configHeader);
+                }
+            }
+
+            if (!validRequestedHeaders.isEmpty()) {
+                response.headers().set(HttpHeaders.ACCESS_CONTROL_ALLOW_HEADERS, String.join(",", validRequestedHeaders));
             }
         }
     }
 
-    private void processMethods(HttpServerResponse response, String requestedMethods) {
+    private void processMethods(HttpServerResponse response, String allowMethodsValue) {
         if (corsConfig.methods.isEmpty()) {
-            response.headers().set(HttpHeaders.ACCESS_CONTROL_ALLOW_METHODS, requestedMethods);
+            response.headers().set(HttpHeaders.ACCESS_CONTROL_ALLOW_METHODS, allowMethodsValue);
         } else {
+            List<String> requestedMethods = Arrays.asList(allowMethodsValue.split(COMMA_SEPARATED_SPLIT_REGEX));
 
-            final String validRequestedMethods = corsConfig.methods
-                    .stream()
-                    .filter(method -> requestedMethods.contains(method.name()))
-                    .map(HttpMethod::name)
-                    .collect(Collectors.joining(","));
+            List<String> validRequestedMethods = new ArrayList<>();
+            for (HttpMethod configMethod : corsConfig.methods) {
+                if (requestedMethods.contains(configMethod.name())) {
+                    validRequestedMethods.add(configMethod.name());
```

Comment:
Shouldn't we be case insensitive too?

Reply:
Not according to the RFC 2616: https://tools.ietf.org/html/rfc2616#section-5.1.1

> The method is case-sensitive.

Reply:
OK, I would have been more lenient but I can live with it :).

Reply:
I'm fine with being lenient :) I'll change this while adding the regexp compilation.